### PR TITLE
Simplify Maven execution to just use IntelliJ bundled maven.

### PR DESCRIPTION
**Pull request by arangatang**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Description
Previously we executed these commands sequentially:
1. `<cmd> clean install`
2. `<cmd> copy-dependencies`

Using <cmd> as:

1. `./mvnw` or `./mvnw.cmd`
2. `./mvn`
3. Bundled IntelliJ Maven Runner

However invoking the commands this way made it possible that:

1. neither `mvn clean install` nor `mvn copy-dependencies` run successfully 
2. `mvn clean` could succeed while `mvn install` could fail
3. `mvn clean install` could succeed while `mvn copy dependencies` could fail

In all these cases we could not upload dependencies and errors in these steps were hidden from users. 
Some of these errors stem from issues with different versions of Java used by users `mvn` installation vs the `jdk` version selected in IntelliJ. 

## What this change does 
We simplify the way we invoke Maven in this PR by:

1. Only leverage the built in maven runner and not that `mvn` or `mvnw` are present
2. Split commands into three seperate steps; `clean`, `install`, `copy-dependencies` to simplify debugging for users
3. Cancel job on failure in any of these steps
4. Fixes JAVA_HOME issues with `mvn` or `mvnw` as sideffect
5. Move logic for invoking Maven from the cluttered `CodeModernizerSessionContext` to its own class.

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt [hunk 1]
```diff
@@ -78,17 +78,16 @@ class CodeModernizerSession(
...
-                is ZipCreationResult.Missing1P -> {
-                    notifyStickyInfo(
-                        message("codemodernizer.notification.info.maven_failed.title"),
-                        message("codemodernizer.notification.info.maven_failed.content")
-                    )
-                    result.payload
-                }
 
-                is ZipCreationResult.Succeeded -> result.payload
+            if(result is ZipCreationResult.Missing1P){
```
Comment (damntrecky) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt:82]: New behavior: If maven fails we do NOT want the users to move forward.

To help increase efficacy of acceptance rates, we need to make sure there is more validation on the front end.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunManager.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            sourceFolder.absolutePath,
+            null,
+            listOf("clean"),
+            emptyList<String>(),
+            null
+        )
+        val cleanTransformRunnable = TransformRunnable()
+        runInEdt {
+            try {
+                transformMavenRunner.run(cleanParams, mvnSettings, cleanTransformRunnable)
```
Comment (damntrecky) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunManager.kt:46]: Change prefix to `Maven Clean:`

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunManager.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            copyCommandList,
+            emptyList<String>(),
+            null
+        )
+        val copyTransformRunnable = TransformRunnable()
+        runInEdt {
+            try {
+                transformMavenRunner.run(copyParams, mvnSettings, copyTransformRunnable)
+            } catch (t: Throwable) {
+                val error = "Maven copy dependencies: Unexpected error when executing bundled Maven copy dependencies"
```
Comment (damntrecky) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunManager.kt:103]: I believe we are using the prefix `Maven Copy:` to be consistent with VSCode.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunManager.kt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            listOf("clean"),
+            emptyList<String>(),
+            null
+        )
+        val cleanTransformRunnable = TransformRunnable()
+        runInEdt {
+            try {
+                transformMavenRunner.run(cleanParams, mvnSettings, cleanTransformRunnable)
+            } catch (t: Throwable) {
+                val error = "Maven Install: Unexpected error when executing bundled Maven clean"
```
Comment (dhasani23) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunManager.kt:48]: Maven Install or Maven Clean?

**Review by dhasani23 — APPROVED** [review]

**Comment by arangatang** [comment]
Closing in favor of #4135

**Pull request closed without merging**